### PR TITLE
Consume Microsoft.DotNet.RemoteExecutor

### PR DESCRIPTION
**Pull request by ViktorHofer**
Requires https://github.com/dotnet/arcade/pull/2176

Commit 1 contains the infra work
Commit 2 the source changes

The failing arm64 leg is unrelated.

cc @danmosemsft

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@ericstj @safern this is now ready to go in. Please take a look and also check the related one in arcade https://github.com/dotnet/arcade/pull/2176. The sooner I get this in the less conflicts I have to deal with :)

File: Directory.Build.props [hunk 1]
```diff
@@ -148,6 +148,7 @@
        -->
   <PropertyGroup>
     <RestoreSources Condition="'$(DotNetBuildOffline)' != 'true'">
+      https://www.myget.org/F/vihofer-staging/api/v3/index.json;
```
Comment (ViktorHofer) [inline Directory.Build.props]: Will be removed when the arcade PR is in.

File: Directory.Build.props [hunk 2]
```diff
@@ -389,9 +390,9 @@
     <BinPlaceNETFXRuntime Condition="'$(_bc_TargetGroup)' == 'netfx'">true</BinPlaceNETFXRuntime>
 
     <NETCoreAppTestSharedFxVersion>9.9.9</NETCoreAppTestSharedFxVersion>
-    <TestHostRootPath>$(ArtifactsBinDir)testhost/$(BuildConfiguration)/</TestHostRootPath>
-    <NETCoreAppTestHostFxrPath>$(TestHostRootPath)host/fxr/$(NETCoreAppTestSharedFxVersion)/</NETCoreAppTestHostFxrPath>
-    <NETCoreAppTestSharedFrameworkPath>$(TestHostRootPath)shared/Microsoft.NETCore.App/$(NETCoreAppTestSharedFxVersion)/</NETCoreAppTestSharedFrameworkPath>
+    <TestHostRootPath>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'testhost', '$(BuildConfiguration)'))</TestHostRootPath>
```
Comment (ViktorHofer) [inline Directory.Build.props]: Wrong slashes were causing failures when running uap locally as we use the path in the cmd script with surrounding quotes.

**Comment by ViktorHofer** [comment]
Downgrading Arcade.Sdk version as it causes build errors:

```
D:\a\1\s\.dotnet\sdk\2.2.103\MSBuild.dll /nologo -maxcpucount /m -verbosity:m /v:minimal /bl:D:\a\1\s\artifacts\log\Debug\ToolsetRestore.binlog /clp:Summary /clp:ErrorsOnly;NoSummary /nr:False /p:ContinuousIntegrationBuild=True /p:TreatWarningsAsErrors=true /p:__ToolsetLocationOutputFile=D:\a\1\s\artifacts\toolset\1.0.0-beta.19202.3.txt /t:__WriteToolsetLocation /warnaserror D:\a\1\s\artifacts\toolset\restore.proj
D:\a\1\s\artifacts\toolset\restore.proj : error : restore depends on Microsoft.DotNet.Arcade.Sdk (>= 1.0.0-beta.19202.3) but Microsoft.DotNet.Arcade.Sdk 1.0.0-beta.19202.3 was not found. An approximate best match of Microsoft.DotNet.Arcade.Sdk 1.0.0-dev.18424.13 was resolved.
D:\a\1\s\.dotnet\sdk\2.2.103\MSBuild.dll /nologo -maxcpucount /m -verbosity:m /v:minimal /bl:D:\a\1\s\artifacts\log\Debug\Build.binlog /clp:Summary /nr:False /p:ContinuousIntegrationBuild=True /p:TreatWarningsAsErrors=true /p:Configuration=Debug /p:RepoRoot=D:\a\1\s /p:Restore=True /p:DeployDeps=False /p:Build=False /p:Rebuild=False /p:Deploy=False /p:Test=False /p:Pack=False /p:IntegrationTest=False /p:PerformanceTest=False /p:Sign=False /p:Publish=False /p:build=True /p:ConfigurationGroup=Debug /p:BuildTests=true /p:ArchGroup=x64 /p:TargetGroup=netcoreapp /p:ArchiveTests=Tests /p:OuterLoop=false /p:RuntimeOS=win10 /warnaserror D:\a\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-dev.18424.13\tools\Build.proj
D:\a\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-dev.18424.13\tools\Build.proj(63,5): error : Property 'Projects' must be specified

Build FAILED.

D:\a\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-dev.18424.13\tools\Build.proj(63,5): error : Property 'Projects' must be specified
    0 Warning(s)
    1 Error(s)
```

https://dev.azure.com/dnceng/public/_build/results?buildId=140429

File: src/System.Buffers/tests/System.Buffers.Tests.csproj [hunk 3]
```diff
@@ -2,20 +2,14 @@
   <PropertyGroup>
     <ProjectGuid>{62E2AD5F-C8D0-45FB-B6A5-AED2C77F198C}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <Configurations>netstandard-Debug;netstandard-Release</Configurations>
     <TestRuntime>true</TestRuntime>
+    <IncludeRemoteExecutor>true</IncludeRemoteExecutor>
```
Comment (safern) [inline src/System.Buffers/tests/System.Buffers.Tests.csproj]: This is now so easy 😄

**Pull request merged**